Update consent metrics (#322)
ethyca/fidesdocs: pages/tutorials/consent-management/consent-management-configuration/consent-monitoring.mdx

import { Callout } from 'nextra-theme-docs'
import TutorialKey from 'components/tutorialkey'
import Screenshot from 'components/screenshot'

> # Consent Management: Optional: Monitoring Consent

Reviewer: ```suggestion
# Consent Management: Consent Reporting
```

<TutorialKey duration="10" product="Fides" topic="Consent Management"/>

<Callout emoji=""> This tutorial requires Fides Cloud or Fides Enterprise. For more information, [talk to our solutions team.](https://ethyca.com/book-demo?utm_source=docs-site) </Callout>

> In this tutorial, we'll review the different options available to monitor consent choices.

Reviewer: ```suggestion
In this tutorial, we'll review the different options available to run reports on saved consent preferences.
```

> After reading this, you'll be familiar with the different methods available to track both opt-in and opt-out choices made by your users.

Reviewer: ```suggestion
There are three main options for reporting:
1. Download saved consent preferences from Fides Cloud as a CSV file for manual reporting
2. Query saved consent preferences from the Fides API for automated reporting
3. Integrate consent events from FidesJS into your frontend product analytics for real-time reporting

After reading this, you'll be familiar with the different methods available to track both opt-in and opt-out choices made by your users. 
```

Author: I forgot about the CSV download!

## Prerequisites

For this tutorial you'll need:

> - An understanding on how to [Configure Privacy Notices](/tutorials/consent-management/consent-management-configuration/privacy-notices)
> - A Fides Cloud or Fides Enterprise account
> - A functioning consent implementation

Reviewer: ```suggestion
- A Fides Cloud or Fides Enterprise account
- A functioning consent implementation
- Knowledge about your own preferred reporting tools for CSV analysis, API integrations, or event analytics
```

> 

Reviewer: ```suggestion
## Option 1: Download CSV Reports from Fides
```

Let's add a quick overview of this option to download reports.

## Using the API

> Fides comes with an expansive API that can cover a multitude of use-cases, including tracking [historical privacy preferences](/dev-docs/api/manage/historical-privacy-preferences). These can also be found at the `docs/` page on your Fides instance.

Reviewer: ```suggestion
Fides comes with an expansive API that can cover a multitude of use-cases, including tracking [historical privacy preferences](/dev-docs/api/manage/historical-privacy-preferences).
```

> To collect the fullest picture possible, your data team can set up a pipeline using the `/api/v1/historical-privacy-preferences` endpoint. With opt-in by default consent, it's important to remember to pair this data alongside traffic counts to ensure a full picture!

Reviewer: I think we need a bit more detail here on how to authenticate and execute a query... something like:
1. Get an OAuth Client ID & Secret from your customer success manager
2. Authenticate with `POST /api/v1/oauth/token`
3. Query `GET /api/v1/historical-privacy-preferences`

There's a bit of detail in https://ethyca.com/docs/dev-docs/api/auth about step #1 but it's not very clear

Author: sounds good 👍🏽  I was looking around to see if it had been covered elsewhere (feels like it could/should be and then have it referenced here)

<Screenshot img="/docs/assets/img/tutorials/consent-management/privacy-preferences.png" description="Privacy Preferences API" />

> ## Using event data

Reviewer: ```suggestion
## Option 3: Integrating FidesJS consent events into your event analytics tool
```

> The `FidesUpdated` event is another way to track consent preferences and push data into your data warehouse. The following snippet is an example of how to listen and act when an explicit consent preference is made.

Reviewer: ```suggestion
When running on a user's browser, FidesJS will automatically emit a `FidesUpdated` event to the `window` object with the user's saved consent preferences. You can subscribe to this event to track changes to user consent (e.g. opt-in, opt-out) and other details such as the consent method (e.g. save, dismiss).

```

```

> window.addEventListener("FidesUpdated", (e) => console.log(Fides.consent, e))

Reviewer: ```suggestion
window.addEventListener("FidesUpdated", (event) => {
  const consent = event.detail.consent; // e.g. { marketing: true, analytics: false }
  const extraDetails = event.detail.extraDetails; // e.g. { consentMethod: "save" }
  console.log("Received 'FidesUpdated' event from FidesJS!");
  console.log("Updated user consent preferences:", consent);
  console.log("Extra event details:", extraDetails);
});
```

With the above event listener callback, you will see event metadata logged to the console like this:
```
Received 'FidesUpdated' event from FidesJS!
Updated user consent preferences: {data_sales_and_sharing: true}
Extra event details: {consentMethod: 'save'}
```

You can then integrate this as a "custom event" into your preferred event analytics tool, such as Google Analytics, Mixpanel, Segment, etc. Please refer to the documentation for each of these tools for more detail.

For example, here is how you might publish `FidesUpdated` as a custom event to Google Analytics:
```
window.addEventListener("FidesUpdated", function(event) {
  const consent = event.detail.consent;
  const extraDetails = event.detail.extraDetails;

  // Send custom event to Google Analytics using gtag
  gtag("event", "custom_event_triggered", {
    "consent": consent,
    "extraDetails": extraDetails,
  });
});
```

```

Some examples of the schema include:

- _Fides.consent_: The current consent preferences for the available notices
- _Fides.experience.region_: The region where the notices were served

> - _Fides.fides_meta.consentMethod_: The method of make a privacy preference (e.g. accept, GPC, dismiss)

Reviewer: I don't think we should encourage using the `Fides` object directly; instead let's just use what is copied to the `event.detail`

This just encourages a lot less potential bugginess. See https://github.com/ethyca/fides/blob/main/clients/fides-js/src/lib/events.ts for some developer docs on the detail
